src/codegen_simulator.rs

```diff
@@ -180,7 +180,7 @@ impl CodeGen for CodeGenSimulator {
                 self.collected_properties.push(map_name.clone());
                 if self.envoy_properties.contains(&map_name) {
                     self.collect_envoy_property(map_name);
-                } else {
+                } else if !map_name.is_empty() {
```

Comment:
What about the case where this does not hit? Should add a diagnostic here or add a reason why this does not matter.

Reply:
This doesn't matter.  If it doesn't hit, that means there is no property.  Since this function cycles through all possible properties and makes sure we're collecting them, but there is no property there, we don't need to collect anything.